app-libs/exchange-oracle/src/coingecko.rs

```diff
@@ -49,6 +63,16 @@ impl CoinGeckoClient {
 	}
 }
 
+impl TradingPairId for CoinGeckoClient {
+	fn crypto_currency_id(&mut self, trading_pair: TradingPair) -> Result<String, Error> {
```

Comment:
I think clippy should catch such things, so I'm a little confused. But where is this `self` used? And why does it need to be mutable?

Reply:
You don't use the `TradingPair` at all, but only the `crypto_currency`. Why not use the `crypto_currency` as function input directly? For the `get(key)` function, you only need a reference anyway, so you can directly require a reference input for this function. This will give less computation overhead as you don't need to clone.

Reply:
I will rework this in task https://github.com/integritee-network/teeracle/issues/4